### PR TITLE
Opacity tween between two logos when site is scrolled

### PR DESCRIPTION
**Pull request by aashishvasu**
Hi Jason,

I've added some more cosmetic improvements to the theme for my own use, and I was wondering if you want to include this in the main branch as well. Basically, this adds similar functionality to what exists in the original Quark theme which changes the logo on top of the site when you scroll down, but I modded it to use any image file and not just .svg colour changes. Ive added a couple images for further clarification.

Blueprint configuration

Logo Tween

### CODE REVIEW

File: blueprints.yaml [hunk 1]
```diff
@@ -39,6 +39,28 @@ form:
       accept:
         - image/*
 
+    use_custom_logo_alt:
```
Comment (jasonccox) [inline blueprints.yaml]: Let's get rid of this setting and just use the `custom_logo_alt` - if there's an alternate logo specified, assume that it should be used.

File: blueprints.yaml [hunk 2]
```diff
@@ -39,6 +39,28 @@ form:
...
+      default: 0
+      options:
+        1: PLUGIN_ADMIN.YES
+        0: PLUGIN_ADMIN.NO
+      validate:
+        type: bool
+    
+    custom_logo_alt:
+      type: file
+      label: Alternate Logo
```
Comment (jasonccox) [inline blueprints.yaml]:
```suggestion
      label: Scrolled Logo
```

File: blueprints.yaml [hunk 3]
```diff
@@ -39,6 +39,28 @@ form:
...
+      description: Toggle to use an alternate logo in the navigation menu when the site is scrolled
+      highlight: 0
+      default: 0
+      options:
+        1: PLUGIN_ADMIN.YES
+        0: PLUGIN_ADMIN.NO
+      validate:
+        type: bool
+    
+    custom_logo_alt:
```
Comment (jasonccox) [inline blueprints.yaml]: I think this naming will be a bit more intuitive.

```suggestion
    custom_logo_scrolled:
```

File: scss/header.scss [hunk 4]
```diff
@@ -26,11 +26,47 @@
 	align-content: center;
 }
 
+/*.header__dual__logo {
```
Comment (jasonccox) [inline scss/header.scss]: Let's get rid of this

File: scss/header.scss [hunk 5]
```diff
@@ -26,11 +26,47 @@
...
 .logo__img {
 	height: 3rem;
 	transition: all 0.5s ease;
 
 	.scrolled & {
 		height: 2rem;
 	}
+}
+
+.logo__img__1 {
```
Comment (jasonccox) [inline scss/header.scss:38]: These class names don't convey much meaning -- how about if we take advantage of the fact that we already put the `scrolled` class on the `body` element when the page is scrolled and put any CSS for the scrolled logo there? (Check out the selector just above this one.)

**Comment by jasonccox** [comment]
This seems like a useful feature to add. I've made a few comments and will take another look after you make those changes. It looks like you'll need to pull my latest changes and recompile the CSS in order to resolve those merge conflicts.

Thanks for contributing your work! Also, I really like your logo!

File: scss/header.scss (hunk 4 above)
Comment (aashishvasu) [reply scss/header.scss]: Done

File: templates/partials/logo.html.twig [hunk 6]
```diff
@@ -10,6 +10,27 @@
   {% set logo_img = media['theme://images/grav-logo.svg'] %}
 {% endif %}
 
+{% if theme_var('custom_logo_scrolled') %}
```
Comment (jasonccox) [inline templates/partials/logo.html.twig:15]: I'm not really understanding the need for so much complexity here. Couldn't we replace this whole `if` block with something like this:
```suggestion
<a href="{{ home_url }}">
  {{ logo_img.html('', config.site.title, 'logo__img')|raw }}
  
  {% if theme_var('custom_logo_scrolled') %}
    {% set logo_img_scrolled = media['image://webfolio/' ~ (theme_var('custom_logo_scrolled')|first).name] %}

    {{ logo_img_scrolled.html('', config.site.title, 'logo__img__scrolled')|raw }}
  {% endif %}
</a>
```

Then the `.scrolled` class can be used to determine which should be shown/hidden.

(Also, the bit above where I check `image://webfolio/` and `theme://images` is just a backwards compatibility thing - no need to do that here.)

**Pull request closed without merging**